eslint-plugin-prettier.js

```diff
@@ -55,6 +61,26 @@ function reportDifference(context, difference) {
   });
 }
 
+/**
+ * get normalized filepath in case of virtual filename
+ * @param {string} filepath
+ * @returns {string}
+ */
+function normalizeFilepath(filepath) {
+  try {
+    if (fs.statSync(filepath).isFile()) {
+      return filepath;
+    }
+  } catch (err) {
+    // https://github.com/eslint/eslint/issues/11989
+    if (err.code === 'ENOTDIR') {
+      return normalizeFilepath(path.dirname(filepath));
```

Comment:
It could be recursive as described at https://github.com/eslint/eslint/pull/14227#discussion_r602802758